profiles/common/modules/custom/nexteuropa_core_views/plugins/component/views_europa_component_plugin_rows.inc

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * @file
+ * Definition of a Views row style plugin.
+ */
+
+/**
+ * Class to define a row plugin handler.
+ *
+ * @codingStandardsIgnoreFile
+ */
+class ViewsEuropaComponentPluginRows extends views_plugin_row {
+
+  /**
+   * @inheritDoc
+   */
+  public function option_definition() {
+    $options = parent::option_definition();
+    $options['theme'] = '';
+    $options['variables'] = array();
+    return $options;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public function options_form(&$form, &$form_state) {
+    parent::options_form($form, $form_state);
+
+    $hooks_theme = variable_get('nexteuropa_core_views_hooks', array());
+    $form['theme'] = array(
+      '#type' => 'select',
+      '#required' => TRUE,
+      '#title' => t('Theme hook'),
+      '#description' => t('TODO'),
+      '#empty_option' => t('<none>'),
+      '#options' => array_combine(array_keys($hooks_theme), array_keys($hooks_theme)),
+      '#default_value' => $this->options['theme'],
+      '#ajax' => array(
+        'path' => views_ui_build_form_url($form_state),
+      ),
+    );
+
+    // Pre-build all of our option lists for the dials and switches that follow.
+    $fields = array();
+    foreach ($this->display->handler->get_handlers('field') as $field => $handler) {
+      if ($label = $handler->label()) {
+        $fields[$field] = $label;
+      }
+      else {
+        $fields[$field] = $handler->ui_name();
+      }
+    }
+
+    $hook = $this->options['theme'];
+    if (isset($form_state['values']['row_options']['theme'])) {
+      $hook = $form_state['values']['row_options']['theme'];
+    }
+
+    if (!$hook) {
+      return;
+    }
+
+    $form['variables'] = array(
+      '#type' => 'fieldset',
+      '#title' => 'Theme hook variables mapping'
+    );
+
+    foreach ($hooks_theme[$hook]['variables'] as $variable) {
+      $form['variables'][$variable] = array(
+        '#type' => 'select',
+        '#empty_option' => t('<none>'),
+        '#title' => t('Theme variable: @variable', array('@variable' => $variable)),
+        '#options' => $fields,
+        '#default_value' => $this->options['variables'][$variable],
+      );
+    }
+  }
+
+  /**
+   * @inheritDoc
+   */
+  function render($row) {
+    static $row_index;
+    if (!isset($row_index)) {
+      $row_index = 0;
+    }
+
+    // In order to have automatic theme hooks suggestions.
+    $theme = sprintf('%s__%s__%s', $this->options['theme'], $this->view->name, $this->view->current_display);
+
+    // We build the variables array based on fields mapping.
+    $variables = array();
+    foreach ($this->options['variables'] as $key => $variable) {
```

Comment:
@drupol what about moving this part into the `pre_render` method? It's not something super important but we should keep rendering as clean as it's possible.